napari/utils/_tests/test_magicgui.py

```diff
@@ -0,0 +1,64 @@
+import sys
+
+import numpy as np
+import pytest
+
+from napari.layers import Image, Labels, Layer, Points
+
+magicgui = pytest.importorskip('magicgui', reason='please install magicgui.')
+
+
+def test_magicgui_returns_image(make_test_viewer):
+    """make sure a magicgui function returning Image adds an Image."""
+
+    @magicgui.magicgui
+    def add_image() -> Image:
+        return np.random.rand(10, 10)
+
+    viewer = make_test_viewer()
+    viewer.window.add_dock_widget(add_image.Gui())
+    assert len(viewer.layers) == 0
+    add_image()  # should add a new layer to the list
+    assert len(viewer.layers) == 1
+    assert viewer.layers[0].name == 'add_image result'
+
+    add_image()  # should just update existing layer on subsequent calls
+    assert len(viewer.layers) == 1
+    assert viewer.layers[0].name == 'add_image result'
+    assert isinstance(viewer.layers[0], Image)
+
+
+def test_magicgui_returns_label(make_test_viewer):
+    """make sure a magicgui function returning Labels adds a Labels."""
+
+    @magicgui.magicgui
+    def add_labels() -> Labels:
+        return np.random.rand(10, 10)
+
+    viewer = make_test_viewer()
+    viewer.window.add_dock_widget(add_labels.Gui())
+    assert len(viewer.layers) == 0
+    add_labels()  # should add a new layer to the list
+    assert len(viewer.layers) == 1
+    assert viewer.layers[0].name == 'add_labels result'
+    assert isinstance(viewer.layers[0], Image)
+
+
+@pytest.mark.skipif(sys.platform == "darwin", reason="segfault on mac CI")
```

Comment:
Was this a known issue, or unexpected?

Reply:
> not sure why, but was getting segfault on one specific test on mac CI (but not locally). just gonna skip it for now.

Ah just saw this, sorry missed it during my review. That's fair enough, we can revisit later

Reply:
We're actually getting some fails on window here too, see https://cirrus-ci.com/task/5190124025675776 on master right now

```
napari/utils/_tests/test_magicgui.py::test_magicgui_returns_layer_tuple Windows fatal exception: access violation
```

I think it's fine to wait till after `0.4.2` release to fix, but wanted to flag so we don't forget. It appears to happen non-deterministically

Reply:
ok, should probably just remove that test for now